Sources/MapboxMaps/Annotations/ViewAnnotation.swift

```diff
@@ -153,35 +171,36 @@ public final class ViewAnnotationManager {
 
         for position in positions {
             validateAnnotation(byAnnotationId: position.identifier)
-            guard let annotationView = annotationViewsById[position.identifier] else {
+            guard let view = viewsById[position.identifier] else {
                 continue
             }
-            annotationView.translatesAutoresizingMaskIntoConstraints = true
-            annotationView.frame = CGRect(
+            view.translatesAutoresizingMaskIntoConstraints = true
+            view.frame = CGRect(
                 origin: position.leftTopCoordinate.point,
                 size: CGSize(width: CGFloat(position.width), height: CGFloat(position.height))
             )
-            annotationView.setInternalVisibility(isHidden: false)
+            view.isHidden = false
+            expectedHiddenForId[position.identifier] = false
             visibleAnnotationIds.insert(position.identifier)
         }
 
-        let annotationsToHide = Set<String>(annotationViewsById.keys).subtracting(visibleAnnotationIds)
+        let annotationsToHide = Set<String>(viewsById.keys).subtracting(visibleAnnotationIds)
         for id in annotationsToHide {
             validateAnnotation(byAnnotationId: id)
-            annotationViewsById[id]?.setInternalVisibility(isHidden: true)
+            viewsById[id]?.isHidden = true
+            expectedHiddenForId[id] = true
         }
     }
 
     internal func validateAnnotation(byAnnotationId id: String) {
-        guard let annotationView = annotationViewsById[id] else { return }
-        // If the user explicitly removed the ViewAnnotation or it's wrapped view
-        // we need to remove it from our layout calculation
-        if annotationView.subviews.isEmpty || annotationView.superview == nil {
-            try? remove(annotationView)
+        guard let view = viewsById[id] else { return }
+        // If the user removed the view from it's superview, then we can remove it from our layout calculation
+        if view.superview == nil {
+            try? remove(view)
         }
```

Comment:
Another option would be for us to log a warning and re-add the view for symmetry with how we handle isHidden.

Reply:
Good idea! Changed it to work that way.